tsl/src/continuous_aggs/materialize.c

```diff
@@ -171,15 +175,27 @@ continuous_agg_materialize(int32 materialization_id, bool verbose)
 
 	/*
 	 * Transaction 2: get the values from the invalidation table,
+	 *                and copy to materialization_invalidation_log
+	 */
+	StartTransactionCommand();
+	PushActiveSnapshot(GetTransactionSnapshot());
+	caggs = ts_continuous_aggs_find_by_raw_table_id(cagg_data.raw_hypertable_id);
```

Comment:
We should discuss with @JLockerman  if this can be done in transaction 1 before the `invalidation_threshold_set` I don't see why it couldn't. I think i'd prefer not to add another transaction if possible, but would like to discuss more broadly. Let's not change until we discuss with Josh tomorrow.